chainer/functions/math/basic_math.py

```diff
@@ -510,24 +598,62 @@ def check_type_forward(self, in_types):
         type_check.expect(in_types[0].dtype.kind == 'f')
 
     def forward(self, x):
+        self.retain_outputs((0,))
         value = _preprocess_const(x[0], self.value)
-        self.y = utils.force_array(value ** x[0])
-        return self.y,
 
-    def backward_cpu(self, x, gy):
-        # TODO(beam2d): Make it not use the input
-        value = _preprocess_const(x[0], self.value)
-        return utils.force_array(
-            numpy.log(value, dtype=x[0].dtype) * self.y * gy[0]),
+        y = value ** x[0]
 
-    def backward_gpu(self, x, gy):
-        value = _preprocess_const(x[0], self.value)
-        gx = cuda.elementwise(
-            'T x, T gy, T value', 'T gx',
-            'gx = log(value) * pow(value, x) * gy',
-            'pow_const_var_bwd')(x[0], gy[0], value)
+        if isinstance(x[0], numpy.ndarray):
```

Comment:
Would you fix this branch as `PowVarVar`?

Reply:
Fixed.